csrc/transformer/transform_kernels.cu

```diff
@@ -259,11 +259,21 @@ __global__ void bias_add_transform_0213<__half>(__half* output,
 
     bias_arr = bias_vec[d3];
     vals_arr = vals_vec[d3];
+    float2 bias_arr_f[4];
+    float2 vals_arr_f[4];
+#pragma unroll
+    for (int l = 0; l < 4; l++) {
+        bias_arr_f[l] = __half22float2(bias_half[l]);
+        vals_arr_f[l] = __half22float2(vals_half[l]);
+        vals_arr_f[l].x += bias_arr_f[l].x;
+        vals_arr_f[l].y += bias_arr_f[l].y;
+        output_half[l] = __float22half2_rn(vals_arr_f[l]);
+    }
 
-    output_half[0] = vals_half[0] + bias_half[0];
-    output_half[1] = vals_half[1] + bias_half[1];
-    output_half[2] = vals_half[2] + bias_half[2];
-    output_half[3] = vals_half[3] + bias_half[3];
+    // output_half[0] = vals_half[0] + bias_half[0];

```

Comment:
Please remove these lines.

Reply:
I was thinking to use this part of the code in another PR and make the half-addition configurable. I am however not sure if that would impact the performance a lot! What do you think?

Reply:
I have add the __ACC_HALF__ in the kernel code, which we can set it later at compile time. Is this fine?